pkg/apis/pipeline/v1beta1/taskrun_defaults_test.go

```diff
@@ -394,28 +340,19 @@ func TestTaskRunDefaulting(t *testing.T) {
 				},
 			},
 		},
-		wc: func(ctx context.Context) context.Context {
-			s := config.NewStore(logtesting.TestLogger(t))
-			s.OnConfigChanged(&corev1.ConfigMap{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: config.GetDefaultsConfigName(),
-				},
-				Data: map[string]string{
-					"default-timeout-minutes": "5",
-					"default-service-account": "tekton",
-					"default-pod-template":    "nodeSelector: { 'label': 'value' }",
-				},
-			})
-			return s.ToContext(ctx)
+		defaults: map[string]string{
+			"default-timeout-minutes": "5",
+			"default-service-account": "tekton",
+			"default-pod-template":    "nodeSelector: { 'label': 'value' }",
 		},
 	}}
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
-			got := tc.in
 			ctx := context.Background()
-			if tc.wc != nil {
-				ctx = tc.wc(ctx)
+			if len(tc.defaults) > 0 {
```

Comment:
Ideally we wouldn't need this check; SetDefaults should work with empty data

Reply:
Yes, it should work. But we do want to use this check to avoid some unnecessary code execution?

Reply:
personally I think readability is more important but it's subjective